syft/source/metadata.go

```diff
@@ -2,6 +2,7 @@ package source
 
 // Metadata represents any static source data that helps describe "what" was cataloged.
 type Metadata struct {
+	ID            string        // the id generated from the parent source struct
```

Comment:
is this value part of the hash function? (since metadata is not ignored explicitly on the parent struct). If so, then I think this may need to be hash ignored